coding-standards-and-naming.rst

```diff
@@ -193,135 +83,61 @@ Bad class names:
-   uses ``Cassandra`` itself.
--  ``couchbase``: Uses ``Couchbase`` namespace, e.g. ``Couchbase\Document``.
--  ``crypto``: Uses ``Crypto`` namespace, e.g. ``Crypto\PBKDF2``.
--  ``decimal``: Uses ``Decimal`` namespace, e.g. ``Decimal\Decimal``.
--  ``ds``: Uses ``Ds`` namespace, e.g. ``Ds\Collection``.
--  ``grpc``: Uses ``Grpc`` namespace, e.g. ``Grpc\Server``.
--  ``http``: Uses ``http`` namespace, e.g. ``http\Client``.
--  ``mongodb``: Uses ``MongoDB`` namespace, e.g. ``MongoDB\Driver\Manager``.
--  ``mosquitto``: Uses ``Mosquitto`` namespace, e.g. ``Mosquitto\Client``.
--  ``mysql_xdevapi``: Uses ``mysql_xdevapi`` namespace, e.g.
-   ``mysql_xdevapi\Collection``.
--  ``parallel``: Uses ``parallel`` namespace, e.g. ``parallel\Runtime``.
--  ``parle``: Uses ``Parle`` namespace, e.g. ``Parle\Lexer``.
--  ``pcov``: Uses ``pcov`` namespace, e.g. ``pcov\start()``.
--  ``pq``: Uses ``pq`` namespace, e.g. ``pq\Connection``.
--  ``rdkafka``: Uses ``RdKafka`` namespace, e.g. ``RdKafka\Producer``. Also uses
-   ``RdKafka`` itself, and a handful of ``rd_kafka_*()`` functions.
--  ``xlswriter``: Uses ``Vtiful\Kernel`` namespace, e.g.
-   ``Vtiful\Kernel\Excel``.
--  ``yaf``: Uses ``Yaf`` namespace, e.g. ``Yaf\Application``. Also supports
-   aliases in the global namespace, e.g. ``Yaf_Application``.
--  ``zstd``: Uses ``Zstd`` namespace, e.g. ``Zstd\compress()``. However, it also
-   declares ``zstd_*()`` functions in the global namespace.
-
-It is notable that with the exception of ``xlswriter``, none of these extensions
-make use of a vendor namespace. They all use the package/extension name as the
-top-level namespace. Some extensions additionally have a global class that
-matches the extension name, e.g. the ffi extension uses both ``FFI`` and
-``FFI\CType``.
-
-Proposal
-========
-
-This RFC proposes to explicitly allow and encourage the use of namespaces for
-bundled PHP extensions, subject to the guidelines laid out in the following:
-
--  Extensions should not use a vendor namespace.
--  The top-level namespace should match the extension name (apart from casing).
--  Namespace names should follow ``CamelCase``.
--  All symbols defined in the extension should be part of the extension's
+-  Required extensions (including ``Core`` and ``standard``). These extensions
+   are always present, and PHP cannot be built without them.
+
+-  Bundled extensions (including ``ctype`` and ``mbstring``). These extensions
+   are part of the php-src distribution, but PHP can be built without them.
+   Bundled extensions can be either enabled or disabled by default.
+
+-  3rd-party extensions (including ``apcu`` and ``igbinary``). These extensions
+   are not part of the php-src distribution, and either available through PECL,
+   or on GitHub.
+
+Extensions may move between these three categories over time. ``hash`` and
+``json`` recently moved from "bundled" to "required" (though I believe
```

Comment:
Recently as of when?  The example should fix a release in which the move happened, for clarity.